broker/src/main/java/io/zeebe/broker/system/management/BrokerAdminServiceImpl.java

```diff
@@ -107,64 +127,77 @@ private void getFollowerPartitionStatus(
   private void getLeaderPartitionStatus(
       final ZeebePartition partition,
       final StreamProcessor streamProcessor,
+      final ExporterDirector exporterDirector,
       final CompletableFuture<PartitionStatus> partitionStatus) {
-    final var positionFuture = streamProcessor.getLastProcessedPositionAsync();
-    positionFuture.onComplete(
-        (processedPosition, positionRetrieveError) -> {
-          if (positionRetrieveError != null) {
-            partitionStatus.completeExceptionally(positionRetrieveError);
-            return;
-          }
+    final var positionFuture = toCompletableFuture(streamProcessor.getLastProcessedPositionAsync());
+    final var currentPhaseFuture = toCompletableFuture(streamProcessor.getCurrentPhase());
+    final var exporterPhaseFuture = toCompletableFuture(exporterDirector.isExportingPaused());
+    final var exporterPosition = exporterDirector.getState().getLowestPosition();
+    final var snapshotId = getSnapshotId(partition);
+    final var processedPositionInSnapshot =
+        snapshotId
+            .flatMap(s -> FileBasedSnapshotMetadata.ofFileName(s))
+            .map(FileBasedSnapshotMetadata::getProcessedPosition)
+            .orElse(null);
+
+    CompletableFuture.allOf(positionFuture, currentPhaseFuture, exporterPhaseFuture)
+        .whenComplete(
+            (nothing, error) -> {
+              if (error != null) {
+                partitionStatus.completeExceptionally(error);
+                return;
+              }
+              final var processedPosition = positionFuture.join();
+              final var processorPhase = currentPhaseFuture.join();
+              final var exporterPhase = exporterPhaseFuture.join() ? "PAUSED" : "EXPORTING";
```

Comment:
Normally it is wise to use enums for that. This would be more future proven. What are you doing if we add a new case like, stopped, canceled, interrupted.

Reply:
Would also be consistent with the processing phase, were we already have that.